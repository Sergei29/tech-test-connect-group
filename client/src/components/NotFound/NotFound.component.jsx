import React from "react";
import PropTypes from "prop-types";

const NotFound = ({ error }) => {
  return (
    <div>
      <h2>Page not found.</h2>
      <br />
      {error ? <p>{error}</p> : null}
    </div>
  );
};

NotFound.defaultProps = {
  error: null,
};

NotFound.propTypes = {
  error: PropTypes.string,
};

export default NotFound;
