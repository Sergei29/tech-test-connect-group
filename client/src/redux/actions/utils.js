/**
 * @description generates the error message out of received error from api
 * @param {Object} error error received from api
 * @param {String} entity name of the entity: 'carList' or 'carCurrent'
 * @returns {String} aggregated error message
 */
export const getErrorMessage = (error, entity) => {
  let errorMessage = "API call failed.";

  if (error.response && error.response.data) {
    errorMessage = error.response.data;
  } else if (error.message) {
    errorMessage = error.message;
  }

  return `${errorMessage}. ${entity}`;
};
