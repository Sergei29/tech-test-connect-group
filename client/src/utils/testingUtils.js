import React from "react";
import { shallow, mount } from "enzyme";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";

const mockStore = configureMockStore([thunk]);

const store = mockStore({
  carList: [
    {
      id: "xe",
      modelYear: "k17",
      url: "/vehicle/xe",
      description:
        "The most advanced, efficient and refined sports saloon that Jaguar has ever produced",
      price: "£30,000",
      media: [{ media: "vehicle", url: "assets/images/xe_k17.jpg" }],
    },
    {
      id: "xf",
      modelYear: "k17",
      url: "/vehicle/xf",
      description:
        "Luxury business saloon with distinctive design, dynamic drive and state-of-the-art technologies.",
      price: "£36,000",
      media: [{ media: "vehicle", url: "assets/images/xf_k17.jpg" }],
    },
  ],
  carCurrent: {
    id: "xe",
    meta: {
      passengers: 5,
      drivetrain: ["AWD", "RWD"],
      bodystyles: ["saloon"],
      emissions: { template: "CO2 Emissions $value g/km", value: 99 },
    },
  },
  error: null,
  loading: false,
});

export const shallowRendererReduxConnected = (ComponentName, props = {}) =>
  shallow(
    <Provider store={store}>
      <BrowserRouter>
        <ComponentName {...props} />
      </BrowserRouter>
    </Provider>
  );

export const mountRendererReduxConnected = (ComponentName, props = {}) =>
  mount(
    <Provider store={store}>
      <BrowserRouter>
        <ComponentName {...props} />
      </BrowserRouter>
    </Provider>
  );

export const shallowRenderer = (ComponentName, props = {}) =>
  shallow(<ComponentName {...props} />);

export const mountRenderer = (ComponentName, props = {}) =>
  mount(
    <BrowserRouter>
      <ComponentName {...props} />
    </BrowserRouter>
  );
