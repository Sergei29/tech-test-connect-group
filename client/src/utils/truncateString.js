export const truncateString = (stringValue, maxLength) => {
  let truncateLocation = stringValue.indexOf(" ", maxLength);
  if (truncateLocation === -1) return stringValue;
  return stringValue.substring(0, truncateLocation) + "...";
};
