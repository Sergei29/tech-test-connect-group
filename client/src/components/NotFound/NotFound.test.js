import { shallowRenderer } from "../../utils/testingUtils";
import NotFound from "./NotFound.component";

describe("NotFound", () => {
  let wrapper, props;

  it("should match the snapshot", () => {
    props = { error: "Resource not found." };
    wrapper = shallowRenderer(NotFound, props);
    expect(wrapper).toMatchSnapshot();
  });

  it("should have heading only if no error message", () => {
    wrapper = shallowRenderer(NotFound);
    expect(wrapper.contains(<h2>Page not found.</h2>)).toBe(true);
    expect(wrapper.contains(<p>Resource not found.</p>)).toBe(false);
  });

  it("should have heading and error displayed if there is error message", () => {
    props = { error: "Resource not found." };
    wrapper = shallowRenderer(NotFound, props);
    expect(wrapper.contains(<h2>Page not found.</h2>)).toBe(true);
    expect(wrapper.contains(<p>Resource not found.</p>)).toBe(true);
  });
});
