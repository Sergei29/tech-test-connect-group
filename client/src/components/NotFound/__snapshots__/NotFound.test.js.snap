// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound should match the snapshot 1`] = `
<div>
  <h2>
    Page not found.
  </h2>
  <br />
  <p>
    Resource not found.
  </p>
</div>
`;
